src/test/util_tests.cpp

```diff
@@ -543,4 +543,30 @@ BOOST_AUTO_TEST_CASE(test_ParseFixedPoint)
     BOOST_CHECK(!ParseFixedPoint("1.", 8, &amount));
 }
 
+BOOST_AUTO_TEST_CASE(util_Logging)
+{
+	{using namespace Logging;
+	BOOST_CHECK_EQUAL(8,sizeof(categoriesEnabled));
+	BOOST_CHECK_EQUAL(0UL,categoriesEnabled);
+	
+	LogTurnOnCategory(THN,true);
+	BOOST_CHECK(LogAcceptCategory(THN));
+	
+	LogTurnOnCategory(THN,false);
+	BOOST_CHECK(!LogAcceptCategory(THN));
+	
+	LogTurnOnCategory(THN,true);
+	LogTurnOnCategory(NET,true);
+	BOOST_CHECK(LogAcceptCategory(THN|NET));
+	
+	LogTurnOnAll(false);
+	BOOST_CHECK_EQUAL(0UL,categoriesEnabled);
+	
+	LogTurnOnAll(true);
+	BOOST_CHECK_EQUAL(ALL,categoriesEnabled);
+
+	BOOST_CHECK_EQUAL(LogGetLabel(ADR),"ADR");
+	}
+}
+
```

Comment:
Additional testing: 
call IsStringTrue() with various inputs and check the output.
call LOG(weird stuff) a bunch of times to exercise it.

Reply:
ok